### PR TITLE
Revert movie matching

### LINKED ISSUE #921: Movies are not identified if flat movie organization is used
**Describe the bug**
In `10.2.0` If flat movie organization is used (`/movies/Movie (year).ext`), Jellyfin fails to identify movies.
It identifies them as "Movies", I guess the parent folder's name.

**Expected behavior**
Prior to `10.2.0` movies were identified correctly if flat movies organization is used.

**Screenshots**
https://imgur.com/a/3fKIMuN

**System (please complete the following information):**
 - OS: Docker
 - Browser: Chrome
 - Jellyfin Version: 10.2.0

**Additional context**
Possible causes: https://github.com/jellyfin/jellyfin/pull/739, https://github.com/jellyfin/jellyfin/pull/621

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
Reverted #933, #924 and #739

**Issues**
Fixes #921

### CODE REVIEW

File: Emby.Naming/Video/VideoListResolver.cs [hunk 1]
```diff
@@ -175,64 +175,71 @@ private IEnumerable<VideoInfo> GetVideosGroupedByVersion(List<VideoInfo> videos)
...
-                // Because of the grouping, we can grab the information from the first movie and make it primary
-                // The remaining movie matches are 'alternate versions'
-                Name = group.First().Name,
-                Year = group.First().Year,
-                Files = group.First().Files,
-                AlternateVersions = group.Skip(1).Select(i => i.Files[0]).ToList(),
-                Extras = group.First().Extras.Concat(group.Skip(1).SelectMany(i => i.Extras)).ToList()
-            });
+                testFilename = testFilename.Substring(folderName.Length).Trim();
+                return testFilename.StartsWith("-", StringComparison.OrdinalIgnoreCase) || Regex.Replace(testFilename, @"\[([^]]*)\]", "").Trim() == string.Empty;
```
Comment (Bond-009) [inline Emby.Naming/Video/VideoListResolver.cs]:
```suggestion
                return testFilename.StartsWith("-", StringComparison.OrdinalIgnoreCase) || string.IsNullOrWhiteSpace(Regex.Replace(testFilename, @"\[([^]]*)\]", string.Empty));
```

Comment (cvium) [reply Emby.Naming/Video/VideoListResolver.cs]: Won't this fail for the empty string?

Comment (Bond-009) [reply Emby.Naming/Video/VideoListResolver.cs]: “Indicates whether a specified string is null, empty, or consists only of white-space characters.”
https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorwhitespace?view=netframework-4.7.2

TLDR: No

Comment (cvium) [reply Emby.Naming/Video/VideoListResolver.cs]: That's misleading...

File: Emby.Naming/Video/VideoListResolver.cs [hunk 2]
```diff
@@ -175,64 +175,50 @@ private IEnumerable<VideoInfo> GetVideosGroupedByVersion(List<VideoInfo> videos)
                 return videos;
             }
 
+            var list = new List<VideoInfo>();
+
             var folderName = Path.GetFileName(Path.GetDirectoryName(videos[0].Files[0].Path));
-            if (!string.IsNullOrEmpty(folderName))
+
+            if (!string.IsNullOrEmpty(folderName) && folderName.Length > 1)
```
Comment (JustAMan) [inline Emby.Naming/Video/VideoListResolver.cs]: Why check for `folderName.Length`?

Comment (cvium) [reply Emby.Naming/Video/VideoListResolver.cs]: Beats me. I'm scared to change it...

Comment (joshuaboniface) [reply Emby.Naming/Video/VideoListResolver.cs]: Yea, if it's a revert, let's avoid changing what the revert does - that'll get us back to known-working code and can go through the finer points later.
